internal/service.go

```diff
@@ -216,21 +220,24 @@ func (h *HauserService) ProcessFilesByDay(exports []client.ExportMeta) (int, err
 	return len(processedBundles), nil
 }
 
-func (h *HauserService) LoadBundles(filename string, bundles ...client.ExportMeta) error {
-	var objPath string
-	var err error
-	if objPath, err = h.warehouse.UploadFile(filename); err != nil {
-		log.Printf("%s", h.warehouse.GetUploadFailedMsg(filename, err))
+func (h *HauserService) LoadBundles(ctx context.Context, filename string, bundles ...client.ExportMeta) error {
+	f, err := os.Open(filename)
+	if err != nil {
 		return err
 	}
-
-	if h.warehouse.IsUploadOnly() {
-		return nil
+	_, objName := path.Split(filename)
+	if err = h.storage.SaveFile(ctx, objName, f); err != nil {
+		log.Println(err)
+		return err
+	}
+	if h.config.StorageOnly {
+		return h.storage.SaveSyncPoints(ctx, bundles...)
 	}
 
-	defer h.warehouse.DeleteFile(objPath)
+	defer h.storage.DeleteFile(ctx, objName)
 
-	if err := h.warehouse.LoadToWarehouse(objPath, bundles...); err != nil {
+	objRef := h.storage.GetFileReference(objName)
```

Comment:
Maybe `SaveFile` above should return the file reference? Then you can just pass that straight to `LoadToWarehouse` rather than needing this intermediate method.